### PR TITLE
feat: Swap RecordingPlayer to LemonModal

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

The `SessionPlayerDrawer` is in an Antd Modal and has some less-than-perfect behaviour around showing/hiding. Thought I'd jump in and fix as I noticed while working on the PersonsModal

## Changes

* Changes `SessionPlayerDrawer` to use LemonModal as well as adding scrolling if too small
  * **NOTE**: The scrolling is not ideal and should be fixed properly at some point to ensure that the modal is usable on small screens. This fix is better than no scroll however.
* Also fixes LemonModal backdrop blur for safari
* Also fixes CloseButton having a higher zindex to make sure it is always above the content

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Opened from the recordings page and the PersonsModal

### CODE REVIEW

File: frontend/src/lib/components/LemonModal/LemonModal.scss [hunk 1]
```diff
@@ -33,6 +36,7 @@
     margin: 1rem auto;
     border-radius: var(--radius);
     background-color: #fff;
+    border: 1px solid var(--border);
```
Comment (benjackwhite) [inline frontend/src/lib/components/LemonModal/LemonModal.scss:39]: This is in the design but was missed out. Noticeably better if the blur isn't supported

File: frontend/src/lib/components/LemonModal/LemonModal.scss [hunk 2]
```diff
@@ -16,11 +17,13 @@
     &.ReactModal__Overlay--after-open {
         background-color: rgba(0, 0, 0, 0.2);
         backdrop-filter: blur(var(--modal-backdrop-blur));
+        -webkit-backdrop-filter: blur(var(--modal-backdrop-blur));
```
Comment (benjackwhite) [inline frontend/src/lib/components/LemonModal/LemonModal.scss:20]: Safari amazingly needs this still

Comment (Twixes) [reply frontend/src/lib/components/LemonModal/LemonModal.scss:20]: Interesting, don't we have autoprefixer for this?

Comment (benjackwhite) [reply frontend/src/lib/components/LemonModal/LemonModal.scss:20]: I wondered this as well but apparently not as this problem is in production too

**Review by alexkim205 — APPROVED** [review]
Thanks for updating to LemonModal! Made a few more fixes:

1. Add LemonDivider between window resolution row and player.
2. Make modal header stick to top. It felt strange that the header scrolled with the player in small height scenarios.

https://user-images.githubusercontent.com/13460330/187730574-02d6e70d-5021-4aa0-acac-c32ae1dc58cc.mov

**Comment by benjackwhite** [comment]
@alexkim205 Nice thinking! I went for a different implementation to take advantage of the LemonModal's built in flexing rather than relying on `sticky` (browser support is always ifffy and it overrode the close button)

**Pull request merged**